### ISSUE TITLE
Merge abc label and title in same location?

### ISSUE THREAD

**Issue by lacstorm** [lukelbd/proplot#201]
### Description

In some circumstance, I would like the abc style as (a)ctrl and (b)no**a**nthro. (It's index + experiment name)
But I now I would get (a)ctrl and (b)no**b**nthro since we have the code `text = abcstyle.replace('a', text).replace('A', text.upper())` to replace all letter 'a' in your abcstyles.

How about we just replace the first letter 'a' in abcstyle such as `text = abcstyle.replace('a', text, 1).replace('A', text.upper())`, so we can have more free abcstyle.
### Steps to reproduce

A "[Minimal, Complete and Verifiable Example](http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports)" will make it much easier for maintainers to help you.

```python
import proplot as plot
fig, axs = plot.subplots(nrows=1, ncols=2)
axs.format(suptitle='Subplot labels demo')
cases = ['ctrl','noanthro']
for i, ax in enumerate(axs):
    ax.format(abc=True,abcstyle='(a).'+cases[i])
```

**Expected behavior**: 
![expected](https://user-images.githubusercontent.com/35789946/85913163-e0718d80-b864-11ea-8676-8e058ece4cb7.png)

**Actual behavior**: [What actually happened]

![default](https://user-images.githubusercontent.com/35789946/85913165-e36c7e00-b864-11ea-94f6-fa09e46f716c.png)

### Proplot version

0.6.4

**Comment by lukelbd** [comment]
Thanks for the report -- you're right, I definitely should replace only the *first* `a` that appears in `abcstyle`.

In the next proplot version, I'd also like to permit putting the `abc` and `title` in the same location and have proplot naturally merge them as in your example. For example, instead of changing `abcstyle` for each subplot, you would use `axs.format(abcloc='left', titleloc='left', abcstyle='(a)')`, set the title for each plot, and the abc labels + titles would be combined rather than the abc label just disappearing (which is the current behavior).

**Comment by lacstorm** [comment]
@lukelbd 
Thanks for your reply. That's a nice improvement. I was wondering yesterday if I could put the abc and title together yesterday. And it turned out that ther overlapped together, XD. Glad that we could do this in the next version. Thanks again for developing such a wonderful package!

**Comment by lukelbd** [comment]
Fixed by 31aca47 -- now proplot matches the case of the *first* `a` or `A` in the string, and only replaced that one:

![fixed](https://user-images.githubusercontent.com/19657652/123888781-770dda00-d911-11eb-9a39-66caff96c7ef.png)

Though will keep this issue open to remind me to address the "merge a-b-c and title in same location" issue.

**Comment by lukelbd** [comment]
This was resolved by 402214f9. Now titles and a-b-c labels in the same location are automatically offset from each other, no matter which location they are in.

Quick example:


```python
import proplot as plot
fig, axs = plot.subplots(ncols=3, refwidth=1.8)
axs.format(abc=True, titleloc='l', title='Title', abcstyle='A.')
axs[1].format(titleloc='uc', abcloc='uc')
axs[2].format(titleloc='r', abcloc='r')
```

![sample](https://user-images.githubusercontent.com/19657652/125189495-994b0600-e1f5-11eb-8a45-3f35a54e8fad.png)
